docs/optimizing.md

```diff
@@ -0,0 +1,125 @@
+# Optimizing Playbooks
+
+By using all of the following optimizations together its estimated that playbook execution time can be reduced by up to 60% compared to default behavior.
+
+## Using the ACI HTTPAPI plugin
+
+The Ansible ACI HTTPAPI plugin instructs Ansible how to interact with an APIC's HTTP based API and execute tasks on the APIC.
+
+### Benefits
+
+- The ACI login credentials and ansible variables can stay in the inventory.
+- Logs in once and executes subsequent tasks without requiring additional logins.
+- Automatically refreshes a login if the token expires during the playbook.
+- Assists with overcoming rate limiting on logins.
+- Leverages APIC's high availability by allowing a list of APIC hosts to be defined as a single ansible host.
+
+### Enabling the plugin
+
+The httpapi plugin can be enabled by setting the following variables:
+
+```ini
+ansible_connection=ansible.netcommon.httpapi
+ansible_network_os=cisco.aci.aci
+```
+
+Instead of using `hostname`, `username` & `password` in the playbook, the following variables can be used in the inventory.
+
+```ini
+ansible_user=admin
+ansible_password="SomeSecretPassword"
+```
+
+The `ansible_host` variable can contain one or more APIC hosts separated by a comma. If multiple hosts are defined the plugin will try executing tasks on the hosts in the order listed until one completes or they all fail.
+
+```ini
+single_apic  ansible_host=apic.host
+cluster_apic ansible_host=apic1.host,apic2.host,apic3.host
+```
+
+Signature-based authentication can be specified in the inventory.

```

Comment:
Do you want to specify also that login call is skipped when doing signature-based authentication? Or we assume users are aware of this. Also what is the benefit of using the connection plugin when doing signature-based authentication, which I think only benefit is providing the key in inventory? Because the usage should not impact execution times correct?

Reply:
True, using a certificate will bypass the login call and add the signature to the cookies to authenticate. There would be no extra execution time benefit just being able to specify in the inventory. I'll update this slightly.